Bottom: 447e87ce32259e82bd88caded82b6a8eff134d8e
Top:    0572a48104f687c8f24e6a8eb965ec37216920ab
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-13 21:00:24 +0100

Change scroll-wheel bindings to match more GTK applications







---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 90da219..0878b0d 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -1203,18 +1203,18 @@ ghid_port_window_mouse_scroll_cb (GtkWidget * widget,
 
   state = (GdkModifierType) (ev->state);
   mk = ghid_modifier_keys_state (&state);
+
   if (mk == NONE_PRESSED)
+    dy = ghid_port.height * gport->zoom / 40;
+  else if (mk == SHIFT_PRESSED)
+    dx = ghid_port.width * gport->zoom / 40;
+  else if (mk == CONTROL_PRESSED)
     {
       zoom_factor = (ev->direction == GDK_SCROLL_UP) ? 0.8 : 1.25;
       ghid_port_ranges_zoom (gport->zoom * zoom_factor);
       return TRUE;
     }
 
-  if (mk == SHIFT_PRESSED)
-    dy = ghid_port.height * gport->zoom / 40;
-  else
-    dx = ghid_port.width * gport->zoom / 40;
-
   if (ev->direction == GDK_SCROLL_UP)
     {
       dx = -dx;
